components/automate-cli/cmd/chef-automate/ha_cert_show.go

```diff
@@ -0,0 +1,414 @@
+func (c *certShowImpl) certShow(cmd *cobra.Command, args []string, remoteService string) error {
+	if !c.nodeUtils.isA2HARBFileExist() {
+		return status.New(status.InvalidCommandArgsError, AUTOMATE_HA_INVALID_BASTION)
+	}
+
+	config, err := c.nodeUtils.pullAndUpdateConfig(&c.sshUtil, nil)
+	if err != nil {
+		return err
+	}
+	certInfo := c.getCerts(config)
+
+	switch remoteService {
+	case CONST_AUTOMATE:
+		if err := c.validateNode(certInfo.AutomateCertsByIP, CONST_AUTOMATE); err != nil {
+			return err
+		}
+		c.printAutomateCertificates(certInfo)
+	case CONST_CHEF_SERVER:
+		if err := c.validateNode(certInfo.ChefServerCertsByIP, CONST_CHEF_SERVER); err != nil {
+			return err
+		}
+		c.printChefServerCertificates(certInfo)
+	case CONST_POSTGRESQL:
+		if c.nodeUtils.isManagedServicesOn() {
+			return status.New(status.InvalidCommandArgsError, "This command is not supported in Managed Services")
+		}
+		if err := c.validateNode(certInfo.PostgresqlCertsByIP, CONST_POSTGRESQL); err != nil {
+			return err
+		}
+		c.printPostgresqlCertificates(certInfo)
+	case CONST_OPENSEARCH:
+		if c.nodeUtils.isManagedServicesOn() {
+			return status.New(status.InvalidCommandArgsError, "This command is not supported in Managed Services")
+		}
+		if err := c.validateNode(certInfo.OpensearchCertsByIP, CONST_OPENSEARCH); err != nil {
+			return err
+		}
+		c.printOpensearchCertificates(certInfo)
+	default:
+		c.printCertificates(certInfo)
+	}
+
+	return nil
+}
+
+// getCerts returns the certificates from the config
+func (c *certShowImpl) getCerts(config *ExistingInfraConfigToml) certShowCertificates {
+	certInfo := certShowCertificates{}
+
+	certInfo.AutomateRootCert = config.Automate.Config.RootCA
+	certInfo.AutomateCertsByIP = config.Automate.Config.CertsByIP
+
+	certInfo.ChefServerCertsByIP = config.ChefServer.Config.CertsByIP
+
+	if !c.nodeUtils.isManagedServicesOn() {
+		certInfo.PostgresqlRootCert = config.Postgresql.Config.RootCA
+		certInfo.PostgresqlCertsByIP = config.Postgresql.Config.CertsByIP
+
+		certInfo.OpensearchRootCert = config.Opensearch.Config.RootCA
+		certInfo.OpensearchAdminKey = config.Opensearch.Config.AdminKey
+		certInfo.OpensearchAdminCert = config.Opensearch.Config.AdminCert
+		certInfo.OpensearchCertsByIP = config.Opensearch.Config.CertsByIP
+	}
+
+	return certInfo
+}
+
+// printCertificates prints all certificates
+func (c *certShowImpl) printCertificates(certInfo certShowCertificates) {
+	c.printAutomateCertificates(certInfo)
+	c.printChefServerCertificates(certInfo)
+
+	if !c.nodeUtils.isManagedServicesOn() {
+		c.printPostgresqlCertificates(certInfo)
+		c.printOpensearchCertificates(certInfo)
+	}
+}
+
+// printAutomateCertificates prints automate certificates
+func (c *certShowImpl) printAutomateCertificates(certInfo certShowCertificates) {
+	c.writer.Title("Automate Certificates")
+	c.writer.HR()
+
+	c.writer.Println("========================Automate Root CA========================")
```

Comment:
Are we got the approval to show a response like this to the customers?

Reply:
No, we haven't. Please let me know whom to contact for this.